### DISCUSSION TITLE
[Contract Verification] Deployed bytecode is not equal to generated one from given source

### DISCUSSION THREAD

**Discussion by wshino** [zkSync-Community-Hub/zksync-developers#538] [Contract Verification] [answered]
### Environment

Testnet

### Block Explorer

https://explorer.zksync.io/

### Issue Type

- [x] Plugin
- [x] Block Explorer

### Contract Address

0x1A9f79A1e796Cde08fBD39Da6ffFa3bCf6E27121

### Compiler Type

Multi-part contract

### zkSolc Version

1.4.1

### Solc Version

0.8.24

### Contract Name

Account factory

### Contract Code

https://github.com/zkemail/clave-contracts/blob/master/contracts/AccountFactory.sol

### Constructor Arguments

0x0A306bA833C8E071F9363bB32465778BFf22E30c
0x077e2Ce1613a8b7647EC55140efd57C722301f72
0x00100000000000020000000003010019000000600330027000000023033001970001000000310355000000000031035500020000003103550003000000310355000400000031035500050000003103550006000000310355000700000031035500080000003103550009000000310355000a000000310355000b000000310355000c000000310355000d000000310355000e000000310355000f0000003103550000000102200190000000280000c13d0000008002000039000000400020043f0000002602000041000000000202041a000000000300041400000000001003e6000000280430009c000000510000413d0000002b01000041000000800010043f0000002001000039000000840010043f0000000801000039000000a40010043f0000002c01000041000000c40010043f0000002d0100004100000088000104300000000002000416000000000202004b0000004f0000c13d0000001f0230003900000024022001970000008002200039000000400020043f0000001f0230018f00000005043002720000003b0000613d00000000050000190000000506500210000000000761034f000000000707043b000000800660003900000000007604350000000105500039000000000645004b000000330000413d000000000502004b0000004a0000613d0000000504400210000000000141034f00000003022002100000008004400039000000000504043300000000052501cf000000000525022f000000000101043b0000010002200089000000000121022f00000000012101cf000000000151019f0000000000140435000000200130008c0000004f0000413d000000800100043d000000250210009c0000007a0000a13d0000000001000019000000880001043000000000011003e0000000c00330021000000029033001970000002a033001c700000000003103b500000000013103af0000002502200197008600810000040f000000000301001900000060033002700000001f0430018f00000023033001970000000505300272000000670000613d00000000060000190000000507600210000000000871034f000000000808043b00000000008704350000000106600039000000000756004b000000600000413d000000000604004b000000750000613d00000003044002100000000505500210000000000605043300000000064601cf000000000646022f000000000151034f000000000101043b0000010004400089000000000141022f00000000014101cf000000000161019f000000000015043500000060013002100000000102200190000000790000613d000000870001042e00000088000104300000002602000041000000000012041b0000002001000039000001000010044300000120000004430000002701000041000000870001042e00000084002104250000000102000039000000000001042d0000000002000019000000000001042d0000008600000432000000870001042e000000880001043000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000ffffffff00000000000000000000000000000000000000000000000000000001ffffffe0000000000000000000000000ffffffffffffffffffffffffffffffffffffffff360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc0000000200000000000000000000000000000040000001000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000ffffffff000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000008c379a0000000000000000000000000000000000000000000000000000000004f766572666c6f770000000000000000000000000000000000000000000000000000000000000000000000000000000000000064000000800000000000000000ad60dac335e601f83844b63083e639a1dd946233c6e66a17ffbd5e5e543e15fa
0x40C28929fBD647c229F446C21090aD83431FD24E

### Hardhat Verify Plugin Version

2.0.3

### Repo Link (Optional)

https://github.com/zkemail/clave-contracts

### Additional Details

I have the problem for verifying contact on zksync sepolia.

hardhat-zksync failed to verify on zksync sepolia when I try to deploy contracts.
I would like to ignore this issue, but if the bytecode of the artifacts-zk in my local is incorrect there is a problem when using L2ContractHelper.computeCreate2Address due to hardcoded.

**Comment by kiriyaga-txfusion** [comment, marked as answer]
It appears that the contract underwent changes between deployment and the verification, resulting in differing metadata (non-executable part of the bytecode). The ZkSync verifier backend is sensitive to these changes, unlike the behavior on Etherscan. On zkSync, only full verification is supported, not partial, at the moment.

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/131957563/577056db-abf0-47c8-b0d9-b536e61b84d0)

**Comment by wshino** [comment]
@kiriyaga-txfusion Thank you for your investigation.
I've updated some modules, and then it seems to work fine.
This problem has been solved. Thank you so much.

```
    "@matterlabs/hardhat-zksync-chai-matchers": "1.3.0",
    "@matterlabs/hardhat-zksync-deploy": "1.3.0",
    "@matterlabs/hardhat-zksync-solc": "1.1.4",
    "@matterlabs/hardhat-zksync-toolbox": "1.3.0",
    "@matterlabs/hardhat-zksync-verify": "1.4.3",
    "@nomicfoundation/hardhat-chai-matchers": "2.0.6",
```

For example (this is another deployed contract from this repo by me)
https://sepolia.explorer.zksync.io/address/0x693478a3EcF96ed751B0B473d8d0ddb3e48F9930#contract
